server/runtime/src/main/java/org/infinispan/server/ExitStatus.java

```diff
@@ -1,22 +1,53 @@
 package org.infinispan.server;
 
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
 import java.io.Serializable;
 
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.marshall.SerializeWith;
+
 /**
  * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
  * @since 10.0
  **/
+@SerializeWith(ExitStatus.Externalizer.class)
 public class ExitStatus implements Serializable {
-   public enum ExitMode { SERVER_SHUTDOWN, CLUSTER_SHUTDOWN, ERROR }
+   public enum ExitMode {
+      SERVER_SHUTDOWN,
+      CLUSTER_SHUTDOWN,
+      ERROR;
+
+      static ExitMode valueOf(int ordinal) {
+         return ExitMode.values()[ordinal];
```

Comment:
Should cache the result of `values()`, because it creates a new array on each invocation

Reply:
I saw that in other Externalizers, but didn't know if it was overkill given how little this command would be executed. Anyway, I have updated the code to cache the values.